hack/ukify/measure/measure.go

```diff
@@ -0,0 +1,259 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package measure
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/google/go-tpm-tools/simulator"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+
+	"github.com/siderolabs/ukify/constants"
+)
+
+type PCRData struct {
+	SHA1   []bankData `json:"sha1,omitempty"`
+	SHA256 []bankData `json:"sha256,omitempty"`
+	SHA384 []bankData `json:"sha384,omitempty"`
+	SHA512 []bankData `json:"sha512,omitempty"`
+}
+
+type bankData struct {
+	// list of PCR banks
+	PCRS []int `json:"pcrs"`
+	// Public key of the TPM
+	PKFP string `json:"pkfp"`
+	// Policy digest
+	POL string `json:"pol"`
+	// Signature of the policy digest in base64
+	SIG string `json:"sig"`
+}
+
+// signatureData returns the hashed signature digest and base64 encoded signature
+type signatureData struct {
+	Digest          string
+	SignatureBase64 string
+}
+
+// SectionData holds a map of Section to file path to the corresponding section
+type SectionsData map[constants.Section]string
+
+func calculatePCRBankData(pcr int, alg tpm2.Algorithm, sectionData SectionsData, privateKeyFile string) ([]bankData, error) {
+	rsaKey, err := parseRSAKey(privateKeyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// get fingerprint of public key
+	pubKeyFingerprint := sha256.Sum256(x509.MarshalPKCS1PublicKey(&rsaKey.PublicKey))
+
+	sim, err := simulator.Get()
+	if err != nil {
+		return nil, fmt.Errorf("creating tpm2 simulator failed: %v", err)
+	}
+
+	defer sim.Close()
+
+	for _, section := range constants.OrderedSections() {
+		if file, ok := sectionData[section]; ok && file != "" {
+			if err := pcrExtent(sim, pcr, alg, append([]byte(section), []byte("\x00")...)); err != nil {
```

Comment:
this is better:

```suggestion
			if err := pcrExtent(sim, pcr, alg, append([]byte(section), 0)); err != nil {
```

but this might also change the underlying `section` slice... so to be on the safe side we could also do:

```suggestion
			if err := pcrExtent(sim, pcr, alg, append(append([]byte(nil), []byte(section)...), 0)); err != nil {
```